bitcoin/src/address.rs

```diff
@@ -88,6 +89,13 @@ pub enum Error {
     UnrecognizedScript,
     /// Address type is either invalid or not supported in rust-bitcoin.
     UnknownAddressType(String),
+    /// Address's network differs from expected one.
+    NetworkValidation {
+	/// Network that was expected.
+	expected: Network,
+	/// Network on which the address was found to be valid.
+	found: Network
+    }
```

Comment:
FYI I would normally complain about this not being a separate type but I guess people will usually parse and validate the address at roughly same place so merging them seems OK.

Reply:
Originally I started to write it as separate type but when I saw that other parsing-related errors are in the big enum, I added it there, otherwise `Address::from_str("…")?.require_network(…)?` would not be possible.

Reply:
Yeah, we would need a third type which seems too much.